rust/datafusion/src/execution/context.rs

```diff
@@ -278,25 +278,23 @@ impl ExecutionContext {
     pub fn register_table(
         &mut self,
         name: &str,
-        provider: Box<dyn TableProvider + Send + Sync>,
+        provider: Arc<dyn TableProvider + Send + Sync>,
     ) {
         self.state
             .lock()
             .unwrap()
             .datasources
-            .insert(name.to_string(), provider.into());
```

Comment:
The `Box` passed into to `register_table` was already immediately wrapped in an `Arc` -- this PR allows the user of this API to keep a reference to the provider themselves.

Reply:
Perhaps this could be a good time to also make this method return an `Option<Arc<..>>` so users can tell when they register a table over an existing table (consistent with `deregister_table`).

Reply:
That is an excellent idea. I will make that change